eval/ph/data_collection.py

```
@@ -0,0 +1,289 @@
+        if num_modes % 2 != 0 or target_dim % 2 != 0:
+            continue
+        setups.append(Setup(num_modes=num_modes, target_dim=target_dim))
+    return setups
+
+
+def _mean_std(values: list[float]) -> tuple[float, float]:
+    arr = np.asarray(values, dtype=np.float64)
+    return float(arr.mean()), float(arr.std(ddof=0))
+
+
+def collect_pipeline_results(
```

Ruff (0.15.20)

[warning] Too many branches (13 > 12)

(PLR0912)